### ISSUE TITLE
Caching of model scores results in inability to get more then default 10 top tokens

### ISSUE THREAD

**Issue by pershinmr** [bigartm/bigartm#709]
For LDA model there exists method `get_top_tokens(self, num_tokens=10, with_weights=False)`, which calls `self._internal_model.scores.add(...)` that contains a check `if score.name not in self._data:` -- this causes method `get_top_tokens` to behave completely wrong for num_tokens > 10, as this need a re-evaluation of actual scores for tokens, which is not happening. All this results in situations like this:
```
In: lda_artm.get_top_tokens(num_tokens=11)[-1]
Out: []
```

**Comment by ofrei** [comment]
@pershinmr Could you please remove `[-1]` at the end of your command, and copy here the result of `lda_artm.get_top_tokens(num_tokens=11)`?

@MelLain Do you have any comments on this? It seems to be a bug in Python API.

**Comment by pershinmr** [comment]
@ofrei I'd rather explain the nature of the problem in more details: in method `get_top_tokens` there is slicing of cached evaluation results:
```
        for topic_index in range(self.num_topics):
            if not with_weights:
                tokens.append(result.token[global_token_index: (global_token_index + num_tokens)])
            else:
                result_token = result.token[global_token_index: (global_token_index + num_tokens)]
                result_weight = result.weight[global_token_index: (global_token_index + num_tokens)]
                tokens.append(list(zip(result_token, result_weight)))
            global_token_index += num_tokens
```
and I would specifically like to draw your attention to this:
```
result.token[global_token_index: (global_token_index + num_tokens)]
```
the problem occurs when the length of `result.token` is smaller then `self.num_topics * num_tokens`, for example in my case of desired 11 tokens (and say 25 topics) the length of `result.token` is 250 (as it is cached) while we are slicing this list with step of 11 - and, thereby, we hit the end of the list at the topic №23, so that for topics №24 and №25 would be nothing left (that is why output for `[-1]`, the last topic, is `[]`)
Does this clarifies the problem?

P.S. yes, this just a small bug in Python API, nothing more

**Comment by pershinmr** [comment]
I could suggest a solution by removing line 
`if score.name not in self._data:` [https://github.com/bigartm/bigartm/blob/master/python/artm/scores.py#L49](url)
But i'm afraid i'm not sure whether there be any side effects of this

**Comment by ofrei** [comment]
@pershinmr This should be fixed by #710 . May I ask you to test if the fix works?

**Comment by pershinmr** [comment]
@ofrei Just tested it - yes, it was fixed by #710. Thank you!
